src/factory/mastercopyDeployer.ts

```diff
@@ -7,51 +7,115 @@ import { getSingletonFactory } from "./singletonFactory";
  *
  * To get the same address on any chain.
  * @param hre hardhat runtime environment
- * @param mastercopyContractFactory
- * @param args
- * @returns The address of the deployed module mastercopy
+ * @param mastercopyContractFactory mastercopy to deploy
+ * @param args the arguments to pass to the mastercopy's constructor
+ * @returns The address of the deployed module mastercopy or undefined if it was already deployed
  */
 export const deployMastercopy = async (
   hre: HardhatRuntimeEnvironment,
   mastercopyContractFactory: ContractFactory,
   args: Array<any>,
   salt: string
-) => {
+): Promise<string | undefined> => {
   const deploymentTx = mastercopyContractFactory.getDeployTransaction(...args);
+
   if (deploymentTx.data) {
-    await deployMastercopyWithInitData(hre, deploymentTx.data, salt);
+    return await deployMastercopyWithInitData(hre, deploymentTx.data, salt);
+  }
+  throw new Error("No deployment data found");
+};
+
+/**
+ * Compute a module's mastercopy address. Where it is or will be deployed.
+ *
+ * @param hre hardhat runtime environment
+ * @param mastercopyContractFactory mastercopy to get address for
+ * @param args the arguments passed to the mastercopy's constructor
+ * @returns {
+ *  address: string; // the address where the module mastercopy will be deployed or was already deployed
+ *  isDeployed: boolean; // true if the module mastercopy was already deployed on this chain
+ * }
+ */
+export const computeTargetAddress = async (
+  hre: HardhatRuntimeEnvironment,
```

Comment:
It's a newly exported function right? Will this be used by client code?

Also MINOR comments on the name:
It feels a bit weird to call it just computeTargetAddress, since we are also assessing deployment. But indeed, it's hard to find a better name.

I like the simplicity, but this just feels like a pure wrapper around hre.ethers.utils.getCreate2Address

If we can find a better name with same extension in characters, let's change it. Otherwise let's keep the simple name

Reply:
Yes, it will be used by client code for getting the contracts address when `deployMastercopy` returns the zero address.